cdap-ui/app/cdap/components/PropertiesEditor/index.js

```diff
@@ -0,0 +1,193 @@
+      .subscribe((res) => {
+        this.setState({
+          systemProperties: res[0].filter((row) => row.key !== 'schema'),
+          userProperties: res[1]
+        });
+      }, (err) => {
+        console.log('Error', err);
+      });
+  }
+
+  fetchUserProperties() {
+    let namespace = NamespaceStore.getState().selectedNamespace;
+    const params = {
+      namespace,
+      entityType: this.props.entityType,
+      entityId: this.props.entityId,
+      scope: 'USER'
+    };
+
+    MyMetadataApi.getProperties(params)
+      .map(convertObjToArr)
+      .subscribe((res) => {
+        this.setState({
+          userProperties: res,
+          activeEdit: null,
+          newValue: ''
+        });
+      });
+  }
+
+  renderSystemProperties() {
+    return this.state.systemProperties.map((row) => {
+      return (
+        <tr key={row.id}>
+          <td>{row.key}</td>
+          <td>{row.value}</td>
+          <td>{T.translate('features.PropertiesEditor.system')}</td>
+          <td></td>
+        </tr>
+      );
+    });
+  }
+
+  renderActions(row) {
+    return (
+      <span>
+        <EditProperty
+          property={row}
+          onSave={this.setEditProperty.bind(this, row)}
+          entityType={this.props.entityType}
+          entityId={this.props.entityId}
+        />
+
+        <DeleteConfirmation
+          property={row}
+          onDelete={this.fetchUserProperties}
+          entityType={this.props.entityType}
+          entityId={this.props.entityId}
+        />
+      </span>
+    );
+  }
+
+  renderUserProperties() {
+    return this.state.userProperties.map((row) => {
+      return (
+        <tr
+          key={row.id}
+          className={classnames({'text-success': row.key === this.state.editedKey})}
+        >
+          <td>{row.key}</td>
+          <td>{row.value}</td>
+          <td>{T.translate('features.PropertiesEditor.user')}</td>
+          <td className="actions">
+            {this.renderActions(row)}
+          </td>
+        </tr>
+      );
+    });
+  }
+
+  setEditProperty(row) {
+    this.setState({ editedKey: row.key });
+
+    this.fetchUserProperties();
+
+    setTimeout(() => {
+      this.setState({editedKey: null});
+    }, 3000);
+  }
+
+  render() {
+    return (
+      <div className="properties-editor-container">
+        <table className="table">
+          <thead>
+            <tr>
+              <th className="key">{T.translate('features.PropertiesEditor.name')}</th>
+              <th className="value">{T.translate('features.PropertiesEditor.value')}</th>
+              <th className="scope">{T.translate('features.PropertiesEditor.scope')}</th>
+              <th className="actions"></th>
+            </tr>
+          </thead>
+
+          <tbody>
+            {this.renderUserProperties()}
+            {this.renderSystemProperties()}
+          </tbody>
+        </table>
+
+        <AddPropertyModal
```

Comment:
We should probably have `Add Property` at the top. Adding more properties will push the button down and it will become invisible.